File: tests/indexing/parsing/lang/fixtures/csharp/Exceptions.cs
using System;

namespace Control;

public class Loader
{
    public string Read(string path)
    {
        try
        {
            string raw = Fetch(path);
            return raw.Trim();
        }
        catch (FileNotFoundException e)
        {
            Log(e);
            return "";
        }
        catch (IOException e) when (e.InnerException != null)
        {
            Log(e.InnerException);
            throw;
        }
        finally
        {
            Cleanup();
        }
    }
}

File: tests/indexing/parsing/lang/fixtures/cpp/Exceptions.cpp
std::string load(const std::string &path) {
    try {
        std::string raw = fetch(path);
        std::string trimmed = trim(raw);
        return trimmed;
    } catch (const std::out_of_range &e) {
        log(e.what());
        return "";
    } catch (const std::exception &e) {
        log(e.what());
        throw;
    }
    return "";
}
